src/pages/Dashboard.tsx

```diff
@@ -176,6 +154,50 @@ const Dashboard = () => {
         };
     };
 
+    const [isTaskModelOpen, setIsTaskModelOpen] = useState<boolean>(false);
+    const handleOpenTaskModal = () => setIsTaskModelOpen(true);
+    const handleCloseTaskModal = () => setIsTaskModelOpen(false);
+
+    const [currentTaskModalMode, setCurrentTaskModalMode] = useState<ModalMode>('Create');
+
+    const [currentEditTaskId, setCurrentEditTaskId] = useState<string>('');
+    const [currentEditTask, setCurrentEditTask] = useState<Task | undefined>();
+
+    const [loadingTaskModal, setLoadingTaskModal] = useState<boolean>(false);
+
+    const generateEditTaskHandler = (taskId: string, task: Task): (() => void) => {
```

Comment:
I think I get what generate means here now, you don't need to label the type of function though. This can just be editTaskHandler or editTask. I don't think you need to return a function here either